### PR TITLE
Modernize the C++ app template

### PR DESCRIPTION
**Pull request by swift-kim**
The C++ template will look similar to the C# template when done.

### CODE REVIEW

**Comment by swift-kim** [comment]
Implementing a simple client wrapper for Tizen is done, which only contains the `FlutterApp` class for now.

Any existing native project should still work unless it has Tizen plugin dependencies. It is recommended to re-generate the project by removing `tizen` directory and running `flutter-tizen create` though.

Native apps cannot target both Tizen 4.0 and Tizen 5.5+ at the same time so only Tizen 5.5+ is support by the native template right now. (c.f. .NET apps choose the right embedder for the current platform version and load it dynamically at runtime.)

File: embedding/cpp/flutter_app.cc [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    return FlutterDesktopGetPluginRegistrar(handle, plugin_name.c_str());
+  }
+  return nullptr;
+}
+
+void FlutterApp::ParseEngineArgs() {
+  char *app_id;
+  if (app_get_id(&app_id) != 0) {
+    return;
+  }
```
Comment (pkosko) [inline embedding/cpp/flutter_app.cc:181]: Maybe adding error log will be valuable for debugging this case? (I didn't compile below suggestion, just written it here by hand).
```suggestion
  int ret = app_get_id(&app_id);
  if (ret != APP_ERROR_NONE) {
    TizenLog::Error("Error gathering application id: %d (%s)", ret, get_error_message(ret));
    return;
  }
```

**Review by pkosko — COMMENTED** [review]
Minor suggestion only.

File: embedding/cpp/flutter_app.cc (hunk 1 above)
Comment (swift-kim) [reply embedding/cpp/flutter_app.cc:181]: Thanks! I'll fix.

File: embedding/cpp/flutter_app.cc [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                                   &window_prop.width) != 0 ||
+      system_info_get_platform_int("http://tizen.org/feature/screen.height",
+                                   &window_prop.height) != 0) {
+    TizenLog::Error("Could not obtain the screen size.");
+    return false;
+  }
+
+  std::string res_path;
+  {
+    auto path = app_get_resource_path();
```
Comment (bbrto21) [inline embedding/cpp/flutter_app.cc:31]: app_get_resource_path can return null pointer. I think it would be better to check the return value.

Comment (swift-kim) [reply embedding/cpp/flutter_app.cc:31]: Thanks!

**Review by bbrto21 — APPROVED** [review]
LGTM!

File: embedding/cpp/flutter_app.cc (hunk 1 above)
Comment (bbrto21) [reply embedding/cpp/flutter_app.cc]: @swift-kim is it done?

Comment (swift-kim) [reply embedding/cpp/flutter_app.cc]: Done!

**Pull request merged**